app/controllers/PsrVersionsController.scala

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import cats.implicits.toShow
+import controllers.actions.{AllowAccessActionProvider, DataRequiredAction, DataRetrievalAction, IdentifierAction}
+import models.SchemeId.Srn
+import models.requests.DataRequest
+import models.{DateRange, PsrVersionsResponse}
+import navigation.Navigator
+import pages.WhichTaxYearPage
+import play.api.i18n.{I18nSupport, MessagesApi}
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents, Result}
+import services.{PsrVersionsService, TaxYearService}
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendBaseController
+import utils.DateTimeUtils.localDateShow
+import views.html.PsrReturnsView
+
+import java.time.LocalDate
+import javax.inject.{Inject, Named}
+import scala.concurrent.{ExecutionContext, Future}
+
+class PsrVersionsController @Inject()(
+     override val messagesApi: MessagesApi,
+     @Named("sipp") navigator: Navigator,
+     identify: IdentifierAction,
+     allowAccess: AllowAccessActionProvider,
+     getData: DataRetrievalAction,
+     requireData: DataRequiredAction,
+     view: PsrReturnsView,
+     taxYearService: TaxYearService,
+     psrVersionsService: PsrVersionsService,
+     val controllerComponents: MessagesControllerComponents
+ )(implicit ec: ExecutionContext) extends FrontendBaseController with I18nSupport {
+
+  def onPageLoad(srn: Srn): Action[AnyContent] =
+    identify.andThen(allowAccess(srn)).andThen(getData).andThen(requireData).async { implicit request =>
+      getWhichTaxYear(srn) { taxYear =>
```

Comment:
Should I comment out the getWhichTaxYear method as it would be useful when the mechanism is developed and use hardcoded value for now or should I delete it completely?